### DISCUSSION TITLE
Issues with the loaders new light components.

### DISCUSSION THREAD

**Discussion by SeaverBeaver** [FakeFishGames/Barotrauma#12820] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The new light components added to the weapons loaders, (coilgun loader, chaingun loader, etc,) don't seem to be ordered sequentially:
 
![image](https://github.com/Regalis11/Barotrauma/assets/95244788/212918a6-7ace-41a0-baf9-10a5e8f2f03a)

That third-from-the-right white light doesn't make sense, assuming it's following the green-yellow-red sequence of full to empty.
Also, this change adds 6 light components to _every_ loader. Not a huge deal on it's own, but these things add up. 

In any case, _please_ consider reworking this to be a simple percentage system, like the existing battery. The framework for such a system is already in the .xml, and it would open up a significant avenue for submarine builders. If an ammunition_percentage_out pin was added, sub builders could go crazy with ammo counters and displays:

![image](https://github.com/Regalis11/Barotrauma/assets/95244788/dcf5b1a6-e4eb-4ea6-807f-6b538ecc48d6)

### Reproduction steps

1. Open the sub editor.
2. Spawn a loader of any kind.
3. Link loader to the appropriate hookups for operation.
4. Observe light components.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

Unstable (v1.1.16.0)

### Which operating system did you encounter this bug on?

Windows

**Comment by PanmanS** [comment]
I truly agree that light order doesnt make any sense + having ammo_output pin would be amazing!

**Comment by jigardi** [comment]
I don't know if everything in this suggestion has been addressed, but the ammo out pin seems to have been implemented.
